runtime/vm/CRIUHelpers.cpp

```diff
@@ -1443,6 +1449,43 @@ loadRestoreArguments(J9VMThread *currentThread, const char *optionsFile, char *e
 	return result;
 }
 
+static void
+transitionToDebugInterpreter(J9JavaVM *vm)
+{
+	if (J9JAVAVM_COMPRESS_OBJECT_REFERENCES(vm)) {
+#if defined(OMR_GC_COMPRESSED_POINTERS)
+		vm->bytecodeLoop = debugBytecodeLoopCompressed;
+#endif /* defined(OMR_GC_COMPRESSED_POINTERS) */
+	} else {
+#if defined(OMR_GC_FULL_POINTERS)
+		vm->bytecodeLoop = debugBytecodeLoopFull;
+#endif /* defined(OMR_GC_FULL_POINTERS) */
+	}
+	J9VMThread *walkThread = J9_LINKED_LIST_START_DO(vm->mainThread);
+	while (NULL != walkThread) {
+		VM_VMHelpers::requestInterpreterReentry(walkThread);
+		walkThread = J9_LINKED_LIST_NEXT_DO(vm->mainThread, walkThread);
+	}
+}
+
+static BOOLEAN
+checkTransitionToDebugInterpreter(J9VMThread *currentThread)
+{
+	BOOLEAN result = TRUE;
+	J9JavaVM *vm = currentThread->javaVM;
+	if (NULL != vm->checkpointState.restoreArgsList) {
+		J9VMInitArgs *restoreArgsList = vm->checkpointState.restoreArgsList;
+		if (0 <= FIND_AND_CONSUME_ARG(restoreArgsList, EXACT_MATCH, VMOPT_XINT, NULL)) {
```

Comment:
my apologies, this should have been `VMOPT_XXNODEBUGINTERPRETER`

Reply:
> this should have been VMOPT_XXNODEBUGINTERPRETER

Is it `VMOPT_XXDEBUGINTERPRETER`?

https://github.com/eclipse-openj9/openj9/blob/d3fa3ff21b423ee56ffd878856ba18c9b2538877/runtime/oti/jvminit.h#L348-L349

Reply:
YEs `VMOPT_XXDEBUGINTERPRETER`